plugins/inputs/prometheus/prometheus.go

```diff
@@ -430,12 +500,20 @@ func (p *Prometheus) gatherURL(u URLAndAddress, acc telegraf.Accumulator) error
 		resp, err = uClient.Do(req)
 	}
 	if err != nil {
-		return fmt.Errorf("error making HTTP request to %q: %w", u.URL, err)
+		if setError(err, requestFields, tags) == nil {
+			// Any error not recognized by `set_error` is considered a "connection_failed"
+			setResult("connection_failed", requestFields, tags)
```

Comment:
Capturing the return code and response time of the request is acceptable for a new metric, however, adding all these additional error conditions is not something to start adding to all the plugins. These errors are reported via the the internal metrics already and you can you that or the lack of data as alerts.